bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -371,6 +371,11 @@ func (r *KubeadmConfigReconciler) refreshBootstrapTokenIfNeeded(ctx context.Cont
 
 	secret, err := getToken(ctx, remoteClient, token)
 	if err != nil {
+		if apierrors.IsNotFound(err) && scope.ConfigOwner.IsMachinePool() {
+			log.Info("Bootstrap token secret not found, triggering creation of new token")
+			config.Spec.JoinConfiguration.Discovery.BootstrapToken.Token = ""
```

Comment:
It seems the cleanup of the token here is not needed, since it is added in `recreateBootstrapToken`
```suggestion
```

Reply:
Makes sense. It maybe better to move it to the beginning of the `recreateBootstrapToken` to make it a part of the rotation process, but it is more of a preference (it feels like a part of it).

Otherwise LGTM

/lgtm